Yosemite/YosemiteTests/Tools/EntityListenerTests.swift

```diff
@@ -145,4 +145,47 @@ class EntityListenerTests: XCTestCase {
 
         wait(for: [expectation], timeout: Constants.expectationTimeout)
     }
+
+
+    /// Tests the thread safety of the readOnlyEntity property.
+    ///
+    func test_hread_safety_for_readOnlyEntity() {
+        // Given: A sample storage account is inserted and saved.
+        let storageAccount = storageManager.insertSampleAccount()
+        viewContext.saveIfNeeded()
+
+        // And: An EntityListener is set up with the readOnly entity being the inserted storage account.
+        let listener = EntityListener(viewContext: viewContext, readOnlyEntity: storageAccount.toReadOnly())
+
+        // And: Expectations for multiple threads.
+        let expectation1 = self.expectation(description: "Thread 1")
+        let expectation2 = self.expectation(description: "Thread 2")
+        let expectation3 = self.expectation(description: "Thread 3")
+
+        // When: Multiple threads access the readOnlyEntity property concurrently.
+        DispatchQueue.global().async {
+            for _ in 0..<1000 {
+                _ = listener.readOnlyEntity
+            }
+            expectation1.fulfill()
+        }
+
+        DispatchQueue.global().async {
+            for _ in 0..<1000 {
+                _ = listener.readOnlyEntity
+            }
+            expectation2.fulfill()
+        }
+
+        DispatchQueue.global().async {
+            for _ in 0..<1000 {
+                _ = listener.readOnlyEntity
+            }
+            expectation3.fulfill()
+        }
+
+        // Then
+        wait(for: [expectation1, expectation2, expectation3], timeout: 5.0)
+    }
```

Comment:
I'm not sure about this either. How about using `DispatchQueue.concurrentPerform` like this [suggestion](https://stackoverflow.com/a/75709186)?

Reply:
I tried it, implementing it in this way:

```
func test_thread_safety_for_readOnlyEntity() {
    // Given: A sample storage account is inserted and saved.
    let storageAccount = storageManager.insertSampleAccount()
    viewContext.saveIfNeeded()

    // And: An EntityListener is set up with the readOnly entity being the inserted storage account.
    let listener = EntityListener(viewContext: viewContext, readOnlyEntity: storageAccount.toReadOnly())

    // When: Multiple threads access the readOnlyEntity property concurrently.
    let iterations = 100
    let expectation = self.expectation(description: "Concurrent access")

    DispatchQueue.global(qos: .userInitiated).async {
        DispatchQueue.concurrentPerform(iterations: iterations) { index in
            print("Iteration \(index)")
            _ = listener.readOnlyEntity
        }
        expectation.fulfill()
    }

    // Then: If no crashes or exceptions occurred, the test is considered successful.
    wait(for: [expectation], timeout: 10.0)
    XCTAssert(true)
}
```

But no luck, the test still passes. I would say that it's probably also "normal"? Since it's quite difficult to replicate thread safety issues?